test-packages/e2e-tests/test/on-prem/odata.spec.ts

```diff
@@ -54,3 +76,13 @@ describe('OData OnPrem', () => {
       });
   }, 60000);
 });
+
+describe('OData Cloud', () => {
+  it('should go through proxy if proxy authorization header matches', async () => {
```

Comment:
Very nice that we have a E2E test now, could you perhaps also add one more test here where you set the proxu in the `http_proxy` env variable as string and use the `registerDestination` or `mockDestinationService` to put only the 

```ts
Destination = {
  url: e2eDestination?.url,
 name: 'myDest',
  authentication: 'NoAuthentication'
}
```

and the proxy with headers should be added by the fetching methods. There are units tests inthe `proxy-util` but no E2E test I think.